### PR TITLE
chore: update DataFusion, upgrade `arrow` `arrow-flight` and `parquet` to `32.0.0`

### PR DESCRIPTION
**Pull request by tustvold**
Closes #

Update DataFusion and arrow to keep up with upstream changes

### CODE REVIEW

**Comment by alamb** [comment]
This PR is still failing CI, sadly

File: compactor/src/warm.rs [hunk 1]
```diff
@@ -302,205 +302,6 @@ mod tests {
         );
     }
 
-    #[tokio::test]
```
Comment (tustvold) [inline compactor/src/warm.rs:305]: @NGA-TRAN advised just removing this test as compactor1 is no longer in use

Comment (NGA-TRAN) [reply compactor/src/warm.rs:305]: Slack discussion for this https://influxdata.slack.com/archives/C04K1QRSDFG/p1675343331841219

**Comment by alamb** [comment]
The dedup plan failure is likely https://github.com/influxdata/influxdb_iox/issues/6819 -- I will rerun

**Comment by alamb** [comment]
I am looking into some of the test failures too

File: parquet_file/src/metadata.rs [hunk 2]
```diff
@@ -1087,17 +1087,21 @@ mod tests {
...
             .expect("should contain metadata");
+        assert_metadata(&iox_parquet_meta.decode().unwrap());
 
-        // And read the metadata directly from the file metadata returned from
-        // encoding.
+        // Read the metadata directly from the file metadata returned from
+        // encoding, should be the same
         let iox_from_file_meta = IoxParquetMetaData::try_from(file_meta)
             .expect("failed to decode IoxParquetMetaData from file metadata");
-        assert_eq!(iox_from_file_meta, iox_parquet_meta);
```
Comment (alamb) [inline parquet_file/src/metadata.rs:1095]: I am not sure why but the thrift encoded bytes are no longer the same. However, I think the important thing to test is that the metadata that is read back from the both sources is consistent (not that the bytes used to encode it are exactly the same). Thus I updated the test to do this


Screenshot 2023-02-03 at 2 30 50 PM

[left.txt](https://github.com/influxdata/influxdb_iox/files/10582098/left.txt)
[right.txt](https://github.com/influxdata/influxdb_iox/files/10582099/right.txt)

Comment (alamb) [reply parquet_file/src/metadata.rs:1095]: perhaps @crepererum  can take double check this on Monday

Comment (tustvold) [reply parquet_file/src/metadata.rs:1095]: Most likely https://github.com/apache/arrow-rs/pull/3527

**Review by alamb — APPROVED** [review]
Looks good to me but I am biased. I don't want to merge this until Monday anyways so perhaps @crepererum  will have a chance to review it briefly (the changes to parquet_file specifically) as well

**Review by crepererum — APPROVED** [review]
Thank you. I esp. like the error clean-ups :)

**Pull request merged**